### ISSUE TITLE
Can't enter fullscreen in Firefox

### ISSUE THREAD

**Issue by eolognt** [SeleniumHQ/selenium#897]
I'm using Protractor, a framework on top of WebDriver designed to be used with Angular. In my application we use fullscreen and would like to test that you enter fullscreen when clicking a certain button. In Chrome it works, but when run in Firefox the click doesn't have any effect. If I pause before the click and manually click the button, fullscreen is entered. 

Here is a minimalistic example of when it works in Chrome but not in Firefox:
```html
<html ng-app="test">

<head>
    <script src="https://code.angularjs.org/1.4.3/angular.min.js"></script>
    <script>
        angular.module("test", []).controller(function() {});
    </script>
</head>

<body>
    <video style="background-color: red;" height="100" width="100"></video>
    <button onclick="enterFullscreen()">Enter fullscreen</button>
    <script>
        var enterFullscreen = function() {
            var elem = document.querySelector("video");
            if (elem.requestFullscreen) {
                elem.requestFullscreen();
            } else if (elem.msRequestFullscreen) {
                elem.msRequestFullscreen();
            } else if (elem.mozRequestFullScreen) {
                elem.mozRequestFullScreen();
            } else if (elem.webkitRequestFullscreen) {
                elem.webkitRequestFullscreen();
            }
        };
    </script>
</body>

</html>
```

```javascript
describe('fullscreen tests', function () {
    it('should enter fullscreen when clicking button', function () {
        browser.get(...);
        element(by.buttonText('Enter fullscreen')).click();
        browser.sleep(3000); // Expecting fullscreen to be entered
        browser.driver.manage().window().setSize(800, 400); // Exit fullscreen
    });
});
```

In Chrome fullscreen is entered, but in Firefox the button is just active and pressed, but nothing else happens. And what causes this to only happen when WebDriver click the button but not when I do it.

**Comment by dpsxp** [comment]
```html
<html>
<head>
  <meta charset="utf-8">
  <title>Test</title>
</head>
<body>
  <button class="btn">Click</button>
  <div class=".player"></div>
</body>
</html>
```
```js
// code
document
  .querySelector('.btn')
  .addEventListener('click', function () {
    document.querySelector('.player').mozRequestFullScreen();
  });
```
```js
// test
module.exports = {
  'Active fullscreen': function(browser) {
    browser
      .click('.btn');
      .end();
  }
};
```

Got the same error on Firefox 40.0.3 with selenium 2.47.1. Using [nightwatch](http://nightwatchjs.org/) to run my tests.

Additional info Firefox is showing a message `Request for full-screen was denied because Element.mozRequestFullScreen() was not called from inside a short running user-generated event handler.`

If I click manually on `.btn` everything works fine also if I run the same test on Chrome everything works fine.

**Comment by barancev** [comment]
Fixed by commit 0eec81da52ba4dfb16de590e6e82ebafa452416f , to be released in 2.48

**Comment by dpsxp** [comment]
Thanks @barancev .
